action.yaml

```
@@ -105,76 +105,18 @@ inputs:
-        env_write(client_payload, "GITHUB_EVENT_PULL_REQUEST_BASE_REPO_NAME", "pullRequest.base.repo.name", "")
-        env_write(client_payload, "GITHUB_EVENT_PULL_REQUEST_BASE_SHA", "pullRequest.base.sha", "${{ github.event.pull_request.base.sha }}")
-        env_write(client_payload, "GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FORK", "pullRequest.head.repo.fork", "${{ github.event.pull_request.head.repo.fork }}")
-        env_write(client_payload, "GITHUB_EVENT_PULL_REQUEST_HEAD_SHA", "pullRequest.head.sha", "${{ github.event.pull_request.head.sha }}")
-        env_write(client_payload, "GITHUB_EVENT_PULL_REQUEST_NUMBER", "pullRequest.number", "${{ github.event.pull_request.number }}")
-        env_write(client_payload, "GITHUB_REF_NAME", "targetRefName", "${{ github.ref_name }}")
-        env_write(client_payload, "INPUT_ARGUMENTS", "arguments", "${{ inputs.arguments }}")
-        env_write(client_payload, "INPUT_CACHE", "cache", "${{ inputs.cache }}")
-        env_write(client_payload, "INPUT_CACHE_KEY", "cacheKey", "trunk-${{ inputs.cache-key }}-${{ runner.os }}-${{ hashFiles('.trunk/trunk.yaml') }}")
-        env_write(client_payload, "INPUT_CACHE_PATH", "cachePath", "~/.cache/trunk")
-        env_write(client_payload, "INPUT_CHECK_ALL_MODE", "checkAllMode", "${{ inputs.check-all-mode }}")
-        env_write(client_payload, "INPUT_CHECK_MODE", "checkMode", "${{ inputs.check-mode }}")
-        env_write(client_payload, "INPUT_CHECK_RUN_ID", "checkRunId", "${{ inputs.check-run-id }}")
-        env_write(client_payload, "INPUT_DEBUG", "debug", "${{ inputs.debug }}")
-        env_write(client_payload, "INPUT_GITHUB_REF_NAME", "targetRefName", "${{ github.ref_name }}")
-        env_write(client_payload, "INPUT_LABEL", "label", "${{ inputs.label }}")
-        env_write(client_payload, "INPUT_SETUP_CACHE_KEY", "setupCacheKey", "${{ inputs.cache-key }}")
-        env_write(client_payload, "INPUT_SETUP_DEPS", "setupDeps", "${{ inputs.setup-deps }}")
-        env_write(client_payload, "INPUT_TARGET_CHECKOUT", "targetCheckout", "")
-        env_write(client_payload, "INPUT_TARGET_CHECKOUT_REF", "targetCheckoutRef", "")
-        env_write(client_payload, "INPUT_TRUNK_PATH", "trunkPath", "${{ inputs.trunk-path }}")
-        env_write(client_payload, "INPUT_UPLOAD_LANDING_STATE", "uploadLandingState", "false")
-        env_write(client_payload, "INPUT_UPLOAD_SERIES", "uploadSeries", "${{ inputs.upload-series }}")
-
-        # if client_payload is not None and "trunkApiAddress" in client_payload:
-        #   github_env.write(f"TRUNK_API_ADDRESS={client_payload['trunkApiAddress']}\n")
+        mkdir -p .trunk
+        ln -s ${{ github.action_path }}/mask-payload .trunk/mask-payload
+        echo .trunk/mask-payload >>.git/info/exclude
+
+    - name: Mask tokens and setup inputs
+      uses: ./.trunk/mask-payload
+      env:
+        MASK_INPUTS: ${{ toJSON(inputs) }}
+        MASK_GITHUB_EVENT_PR: ${{ toJSON(github.event.pull_request) }}
```

Can we do

```suggestion
    - name: Mask tokens and setup inputs
      uses: ./.trunk/mask-payload
      with:
        use-payload: ${{ inputs.check-mode === 'payload' }}
      env:
        MASK_INPUTS: ${{ toJSON(inputs) }}
        MASK_GITHUB_EVENT_PR: ${{ toJSON(github.event.pull_request) }}
```

and in `mask-payload`, just switch on `if (${{ inputs.use-payload }})` when deciding which one to use